march_hardware/src/EtherCAT/EthercatSDO.cpp

```diff
@@ -26,7 +26,19 @@ int sdo_bit8(uint16_t slave, uint16_t index, uint8_t sub, uint8_t value)
   return received_working_counter;
 }
 
-int sdo_bit16(uint16_t slave, uint16_t index, uint8_t sub, uint16_t value)
+int sdo_bit8_read(uint16_t slave, uint16_t index, uint8_t sub, int* val_size, uint8_t* value)
+{
+  ROS_DEBUG("sdo_bit8: slaveIndex %i, reg 0x%X, subindex %i, value 0x%X", slave, index, sub, value);
+  int received_working_counter = ec_SDOread(slave, index, sub, FALSE, val_size, value, EC_TIMEOUTRXM);
+  if (received_working_counter == 0)
+  {
+    ROS_FATAL("(sdo_bit8) Error occurred when writing, reg 0x%X, sub-index %i, data: 0x%X to slave %i", index, sub,
+              value, slave);
+  }
+  return received_working_counter;
```

Comment:
Shouldn't you return the read value? or is that stored inside the pointer uint8_t* value

Reply:
The read value is stored in value. It is parsed as a reference when calling this function.

Reply:
is it not more useful to convert the function to return a value instead of the value inside the pointer? Calling a function called read and always returning the same variable can cause confusion.

Reply:
If we do that we miss out on the returned working counter. We're not always returning the same value, this is purely dependent on the type of object that we are reading from. Moreover, it is common practice to produce this form of read and write functions in c code (where you parse a pointer and return a status/error value).

Reply:
In c++ it is more convention to use a reference, instead of a pointer (pointers are easier to get wrong)

Reply:
So to clarify, just do this?:
`int sdo_bitx_read(uint16_t slave, uint16_t index, uint8_t sub, int &val_size, uint8_t &value)`

Reply:
correct, I think that should work